trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmAccessionImporter.php

```diff
@@ -0,0 +1,885 @@
+              ->fields($values)
+              ->execute();
+
+            // If the primary key is not available, then the insert failed
+            if (!$result) {
+              $this->logger->error("Insertion of stock property \"@property\" into chado.stockprop failed.", ['@property' => $property]);
+              $this->error_tracker = TRUE;
+              return false;
+            }
+          }
+        }
+
+        // If no records exist, then insert the property as normal
+        else {
+          $values = [
+            'stock_id' => $stock_id,
+            'type_id' => $cvterm_id,
+            'value' => $prop_value
+          ];
+          $result = $this->connection->insert('1:stockprop')
+            ->fields($values)
+            ->execute();
+
+          // If the primary key is not available, then the insert failed
+          if (!$result) {
+            $this->logger->error("Insertion of stock property \"@property\" into chado.stockprop failed.", ['@property' => $property]);
+            $this->error_tracker = TRUE;
+            return false;
+          }
+        }
+      }
+      else {
+        $this->logger->error("Unable to retrieve the cvterm_id of property \"@property\"", ['@property' => $property]);
+        $this->error_tracker = TRUE;
+        return false;
+      }
+    }
+  }
+
+  /**
+   * Loads each synonym into the chado.synonym and chado.stock_synonym
+   * tables. Returns true if the insert was successful.
+   *
+   * @param int $stock_id
+   *   The value of the primary key for the stock record in Chado.
+   * @param string $stock_properties
+   *   The name that is a synonym of the current germplasm. Multiple
+   *   synonyms may be specified, in which case they are expected to
+   *   be separated using a comma or semicolon.
+   * @return boolean
+   *   Returns true if inserting all the properties was successful,
+   *   including if there are no properties
+   */
+  public function loadSynonyms($stock_id, $synonyms, $organism_id) {
+
+    if ($synonyms) {
+      // Separate out multiple synonyms if we have them by either
+      // semicolons or commas. Whitespace is optional
+      $all_synonyms = preg_split("/[;,]\s*/", $synonyms);
+
+      foreach ($all_synonyms as $synonym) {
+        $synonym = trim($synonym);
+        $synonym_type_id = $this->getCVterm('synonym');
+
+        // Check for and load any synonyms to chado.synonym
+        $synonym_query = $this->connection->select('1:synonym', 's')
+          ->fields('s', ['synonym_id'])
+          ->condition('s.name', $synonym, '=')
+          ->condition('s.type_id', $synonym_type_id, '=');
+        $synonym_record = $synonym_query->execute()->fetchAll();
+
+        // Make sure there aren't 2 or more records for this synonym
+        if (sizeof($synonym_record) >= 2) {
+          $this->logger->error("Found more than one synonym for \"@synonym\" in chado.synonym.", ['@synonym' => $synonym]);
```

Comment:
Again it would be good to indicate in this error message who you found. Note: you may want to use `fetchCol()` above instead of `fetchAll()` as then you get an array of all the ids without any objects involved.
```suggestion
          $this->logger->error("Found more than one synonym for \"@synonym\" in chado.synonym (synonym_ids @ids).", ['@synonym' => $synonym, '@ids' => implode(', ' $synonym_ids)]);
```

Reply:
Addressed this in commit 708d2e5 (pointing me to use fetchCol() was really helpful) but.... now that I'm seeing your next comment below, does this also need to be an error instead of a notice? I can see there potentially being the same synonym for 2 different germplasm that are a different genus, for example. Is there a better way to handle that situation?